src/legacy/core_plugins/metrics/server/lib/vis_data/request_processors/annotations/query.js

```diff
@@ -19,21 +19,18 @@
 
 import getBucketSize from '../../helpers/get_bucket_size';
 import getTimerange from '../../helpers/get_timerange';
-export default function query(req, panel, annotation) {
+import { buildEsQuery } from '@kbn/es-query';
+
+export default function query(req, panel, annotation, esQueryConfig, indexPattern) {
   return next => doc => {
     const timeField = annotation.time_field;
-    const {
-      bucketSize
-    } = getBucketSize(req, 'auto');
+    const { bucketSize } = getBucketSize(req, 'auto');
     const { from, to } = getTimerange(req);
 
     doc.size = 0;
-    doc.query = {
-      bool: {
-        must: []
-      }
-    };
-
+    const queries = req.payload.query ? [req.payload.query] : [];
```

Comment:
```suggestion
     const queries = req.payload.query || [];
```

@markov00 we're just getting back into reviewing the PR after fixing that bug in master that I mentioned in our zoom sync. Wanted to let you know that you'll need to fix it for TSVB in this PR as well since the relevant file changed. All you have to do is apply the suggestion above. Our PR is https://github.com/elastic/kibana/pull/27636 in case you want a little more info.